lib/methods/sleepchecktask.cpp

```diff
@@ -42,13 +42,24 @@ void SleepCheckTask::ScriptFunc(const Checkable::Ptr& checkable, const CheckResu
     String output = "Slept for " + Convert::ToString(sleepTime) + " seconds.";
 
     double now = Utility::GetTime();
+	CheckCommand::Ptr command = checkable->GetCheckCommand();
+	String commandName = command->GetName();
 
-    cr->SetOutput(output);
-    cr->SetExecutionStart(now);
-    cr->SetExecutionEnd(now);
+	if (Checkable::ExecuteCommandProcessFinishedHandler) {
+		ProcessResult pr;
+		pr.PID = -1;
+		pr.Output = output;
+		pr.ExecutionStart = now;
```

Comment:
The execution is sleeping in this case, so the execution start is the sleeping start (not end).